anta/tests/routing/bgp.py

```diff
@@ -348,130 +229,83 @@ class VerifyBGPPeersHealth(AntaTest):
-                    _add_bgp_failures(failures=failures, afi=afi, safi=safi, vrf=vrf, issue=peer_issues)
+                # Check if the AFI/SAFI state is negotiated
+                capability_status = get_value(peer, f"neighborCapabilities.multiprotocolCaps.{address_family.eos_key}")
+                if not _check_bgp_neighbor_capability(capability_status):
+                    self.result.is_failure(f"{address_family} - Peer:{peer['peerAddress']} AFI/SAFI state is not negotiated; {format_data(capability_status)}")
 
-        if failures:
-            self.result.is_failure(f"Failures: {list(failures.values())}")
+                # Check the TCP session message queues
+                inq = peer["peerTcpInfo"]["inputQueueLength"]
+                outq = peer["peerTcpInfo"]["outputQueueLength"]
+                if address_family.check_tcp_queues and (inq != 0 or outq != 0):
+                    self.result.is_failure(f"{address_family} - Peer:{peer['peerAddress']} session has non-empty message queues; InQ: {inq}, OutQ: {outq}")
 
 
 class VerifyBGPSpecificPeers(AntaTest):
-    """Verifies the health of specific BGP peer(s).
-
-    It will validate that the BGP session is established and all message queues for this BGP session are empty for the given peer(s).
+    """Verifies the health of specific BGP peer(s) for given address families.
 
-    It supports multiple types of Address Families Identifiers (AFI) and Subsequent Address Family Identifiers (SAFI).
+    This test performs the following checks for each specified address family and peer:
 
-    For SR-TE SAFI, the EOS command supports sr-te first then ipv4/ipv6 (AFI) which is handled automatically in this test.
-
-    Please refer to the Input class attributes below for details.
+      1. Confirms that the specified VRF is configured.
+      2. For each specified peer:
+        - Verifies that the peer is found in the BGP configuration.
+        - Checks that the BGP session is in the `Established` state.
+        - Confirms that the AFI/SAFI state is `negotiated`.
+        - Ensures that both input and output TCP message queues are empty.
+          Can be disabled by setting `check_tcp_queues` to `False`.
 
     Expected Results
     ----------------
-    * Success: If the BGP session is established and all messages queues are empty for each given peer.
-    * Failure: If the BGP session has issues or is not configured, or if BGP is not configured for an expected VRF or address family.
+    * Success: If all checks pass for all specified peers in all address families.
+    * Failure: If any of the following occur:
+        - The specified VRF is not configured.
+        - A specified peer is not found in the BGP configuration.
+        - The BGP session for a peer is not in the `Established` state.
+        - The AFI/SAFI state is not `negotiated` for a peer.
+        - Any TCP message queue (input or output) is not empty for a peer when `check_tcp_queues` is `True` (default).
```

Comment:
note to self - check how it renders in doc

Reply:
Since we moved the nested Input models in a separate module, we no longer see them in the tests documentation.

Before:
![image](https://github.com/user-attachments/assets/5057bb29-3a90-466e-aeba-2e995723e8d9)

Should we add a seperate section in the doc for these models or maybe just a pointer to have them displayed in the tests like today?

Reply:
Fixed in #912. Thanks @gmuloc